fetcher-v2/utils/locals.go

```diff
@@ -0,0 +1,83 @@
+package utils
+
+import (
+	"errors"
+
+	"github.com/gofiber/fiber/v2"
+	"github.com/libp2p/go-libp2p/core/host"
+	"github.com/libp2p/go-libp2p/core/peer"
+)
+
+func GetDiscoverString(c *fiber.Ctx) (string, error) {
```

Comment:
I think we should discuss about whether it is a good idea to use fiber's locals data structure for keeping track of the node state.